### DISCUSSION TITLE
arm_matrix_instance_f32 function

### DISCUSSION THREAD

**Discussion by serkanMzlm** [bitcraze/discussions#389] [Q&A] [answered]
Hello there ,
 The word arm_matrix_instance_f32 is mentioned on your Github page.

What exactly is the function of arm_matrix_instance_f32?
why does it always take 1 for the first parameter?

**Comment by krichardsson** [comment, marked as answer]
This is part of a [library called CMSIS-DSP](https://arm-software.github.io/CMSIS-DSP/v1.11.0/index.html), used for signal processing and such

[Documentation for arm_matrix_instance_f32](https://arm-software.github.io/CMSIS-DSP/v1.11.0/structarm__matrix__instance__f32.html) 

An arm_matrix_instance_f32 holds information about a matrix (dimensions and data) that can be passed to functions for matrix operations. The first parameters is the number of rows, you have probably looked at code that uses vectors, but there are other [examples where it is larger than 1](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/kalman_core/kalman_core.c#L220).

**Reply by serkanMzlm** [reply]
thanks for sharing informations
